src/extension.ts

```diff
@@ -124,6 +125,7 @@ export function activate(context: vscode.ExtensionContext) {
     diagnosticCollection.set(document.uri, diagnostics);
   };
 
+  context.subscriptions.push(vscode.languages.registerDocumentFormattingEditProvider('proto', new Formatter(binaryPath)));
```

Comment:
I'm not sure that this is right, but it's similar to what we see for Go [here](https://github.com/golang/vscode-go/blob/0fa41b1c6e4217cc68a61bc97a7cabd34655e3cb/src/language/registerDefaultProviders.ts#L45).